packages/ui/components/pages/Fuse/Modals/ClaimRewardsModal/PendingTransaction.tsx

```diff
@@ -0,0 +1,129 @@
+import { Box, Button, Flex, Icon, Text, VStack } from '@chakra-ui/react';
+import { SupportedAsset } from '@midas-capital/types';
+import { BsFillCheckCircleFill, BsFillXCircleFill } from 'react-icons/bs';
+import { Address } from 'wagmi';
+
+import { Column } from '@ui/components/shared/Flex';
+import Loader from '@ui/components/shared/Loader';
+import TransactionStepper from '@ui/components/shared/TransactionStepper';
+import { useErrorToast, useSuccessToast } from '@ui/hooks/useToast';
+import { TxStep } from '@ui/types/ComponentPropsType';
+
+export const PendingTransaction = ({
+  activeStep,
+  failedStep,
+  steps,
+  isClaiming,
+  poolChainId,
+  assetPerRewardToken,
+}: {
+  activeStep: number;
+  failedStep: number;
+  steps: TxStep[];
+  isClaiming: boolean;
+  poolChainId: number;
+  assetPerRewardToken: { [rewardToken: string]: SupportedAsset | undefined };
+}) => {
+  const successToast = useSuccessToast();
+  const errorToast = useErrorToast();
+
+  const addToken = async (asset: SupportedAsset) => {
```

Comment:
This would be a great hook!

I guess we have something similar in the other PendingTransaction components?


Make sure to return a `useCallback()` as it's more efficient. Without it every rerender, will create this function a new.

```typescript
const useAddTokenToWallet = ({underlying,symbol})=>useCallback(()=>{
// do stuff here
},[underlying,symbol])
```

Reply:
Created `useAddTokenToWallet` hook